sympy/plotting/series.py

```diff
@@ -73,25 +132,803 @@ class BaseSeries:
     is_generic = False
     # Represent generic user-provided numerical data
 
-    def __init__(self):
-        super().__init__()
+    is_vector = False
+    is_2Dvector = False
+    is_3Dvector = False
+    is_slice = False
+    # Represents a 2D or 3D vector data series
+
+    _N = 100
+    # default number of discretization points for uniform sampling. Each
+    # subclass can set its number.
+
+    def __init__(self, *args, **kwargs):
+        kwargs = _set_discretization_points(kwargs.copy(), type(self))
+        # discretize the domain using only integer numbers
+        self.only_integers = kwargs.get("only_integers", False)
+        # represents the evaluation modules to be used by lambdify
+        self.modules = kwargs.get("modules", None)
+        # plot functions might create data series that might not be useful to
+        # be shown on the legend, for example wireframe lines on 3D plots.
+        self.show_in_legend = kwargs.get("show_in_legend", True)
+        # line and surface series can show data with a colormap, hence a
+        # colorbar is essential to understand the data. However, sometime it
+        # is useful to hide it on series-by-series base. The following keyword
+        # controls wheter the series should show a colorbar or not.
+        self.colorbar = kwargs.get("colorbar", True)
+        # Some series might use a colormap as default coloring. Setting this
+        # attribute to False will inform the backends to use solid color.
+        self.use_cm = kwargs.get("use_cm", False)
+        # If True, the backend will attempt to render it on a polar-projection
+        # axis, or using a polar discretization if a 3D plot is requested
+        self.is_polar = kwargs.get("is_polar", False)
+        # If True, the rendering will use points, not lines.
+        self.is_point = kwargs.get("is_point", False)
+        # some backend is able to render latex, other needs standard text
+        self._label = self._latex_label = ""
```

Comment:
I don't think that it is good to add everything as implicit keyword arguments and comments.
These options should be listed explicitly and documented.

Reply:
That is done on each plotting function, which will be a future PR! [The proof is here](https://sympy-plot-backends.readthedocs.io/en/latest/modules/index.html),